packages/vite/src/node/build.ts

```diff
@@ -639,6 +652,34 @@ export async function build(
         }
       })
 
+      // watch config related files in another chokidar instance to
+      // re-run the build function to start a new rollup watcher
+      if (config.build.watchConfig) {
+        const chokidar = await import('chokidar')
+        const configFileChokidar = chokidar.watch(
+          (
+            [
+              config.configFile,
+              ...config.configFileDependencies,
+              '.env',
+              '.env.*',
```

Comment:
The env files needs to be resolved from `config.envDir` too, instead of `process.cwd()` with `path.resolve`.